rct229/ruleset_functions/get_building_scc_window_wall_ratios_dict.py

```diff
@@ -0,0 +1,108 @@
+GET_BUILDING_SCC_WINDOW_WALL_RATIO_DICT__REQUIRED_FIELDS = {
+    "building": {
+        "$..surfaces[*]": ["area"],
+        "$..subsurfaces[*]": ["classification"],
+    }
+}
+
+
+def get_building_scc_window_wall_ratios_dict(climate_zone, building):
+    """Determines the window to wall ratio for each surface conditioning category
+    in a building
+
+    Parameters
+    ----------
+    climate_zone : str
+        One of the ClimateZone2019ASHRAE901 enumerated values
+    building : dict
+        A dictionary representing a building as defined by the ASHRAE229 schema
+
+    Returns
+    -------
+    dict
+        A dictionary that maps each surface conditioning category to its average
+        window to wall ratio
+    """
+    assert_required_fields(
+        GET_BUILDING_SCC_WINDOW_WALL_RATIO_DICT__REQUIRED_FIELDS["building"], building
+    )
+
+    # Get the conditioning category for all the surfaces in the building
+    scc_dict = get_surface_conditioning_category_dict(climate_zone, building)
+
+    # Initialize total window areas
+    total_res_window_area = ZERO.AREA
+    total_nonres_window_area = ZERO.AREA
+    total_mixed_window_area = ZERO.AREA
+    total_semi_exterior_window_area = ZERO.AREA
+
+    # Initialize total wall areas
+    total_res_wall_area = ZERO.AREA
+    total_nonres_wall_area = ZERO.AREA
+    total_mixed_wall_area = ZERO.AREA
+    total_semi_exterior_wall_area = ZERO.AREA
+
+    # Loop through all the surfaces in the building
+    for surface in find_all("$..surfaces[*]", building):
+        # surface conditioning category
+        scc = scc_dict[surface["id"]]
+
+        if get_opaque_surface_type(surface) == OST.ABOVE_GRADE_WALL:
+            surface_area = surface["area"]
+            surface_window_area = ZERO.AREA
+            for subsurface in find_all("subsurfaces[*]", surface):
+                glazed_area = subsurface.get("glazed_area", ZERO.AREA)
+                opaque_area = subsurface.get("opaque_area", ZERO.AREA)
+                surface_window_area += (
+                    glazed_area + opaque_area
+                    if subsurface["classification"] != DOOR
+                    or (glazed_area > opaque_area)
+                    else ZERO.AREA
+                )
+
+            if scc == SCC.EXTERIOR_RESIDENTIAL:
+                total_res_wall_area += surface_area
+                total_res_window_area += surface_window_area
+            elif scc == SCC.EXTERIOR_NON_RESIDENTIAL:
+                total_nonres_wall_area += surface_area
+                total_nonres_window_area += surface_window_area
+            elif scc == SCC.EXTERIOR_MIXED:
+                total_mixed_wall_area += surface_area
+                total_mixed_window_area += surface_window_area
+            elif scc == SCC.SEMI_EXTERIOR:
+                total_semi_exterior_wall_area += surface_area
+                total_semi_exterior_window_area += surface_window_area
+            else:
+                assert scc == SCC.UNREGULATED
```

Comment:
My logic was that if we get to this point, scc must be SCC.UNREGULATED because it is the only other SCC option. The assertion is intended to make that clear (I could add a comment too). If the assertion fails, it implies a bug in our code - probably because get_opaque_surface_type added another type that we are not handling here.